### PR TITLE
[feat] Add serve

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR add serve functionalities to Flash.

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,6 +117,18 @@ predictions = model.predict([
 print(predictions)
 ```
 
+### Serving
+
+Note: `Serve` is a framework agnostic serving engine ! [Learn more](flash_examples/serve/generic/boston_prediction/inference_server.py)
```
Comment (luiscape) [inline README.md:122]: Remove `Note:` 

```suggestion
`Serve` is a framework agnostic serving engine ! [Learn more](flash_examples/serve/generic/boston_prediction/inference_server.py)
```

File: README.md [hunk 2]
```diff
@@ -117,6 +117,18 @@ predictions = model.predict([
...
+
+```python
+from flash import FlashServeModel
```
Comment (luiscape) [inline README.md:125]: I suggest changing to `Serve`  -- it sounds more elegant than `FlashServeModel`. Then with the proper namespacing we can know what happens easily:

```python
import flash

flash.Serve()
```

File: flash/core/data/batch.py [hunk 3]
```diff
@@ -251,6 +317,8 @@ def forward(self, batch: Sequence[Any]):
             else:
                 self.save_fn(final_preds)
         else:
+            if self.is_serving and isinstance(final_preds, list) and len(final_preds) == 1:
```
Comment (luiscape) [inline flash/core/data/batch.py:320]: Can you add an in-line comment on why this needs to be done?

**Review by luiscape — COMMENTED** [review]
The major change I suggest is on the naming of `FlashServeModel` to just `Serve`. It's more elegant and in-line with other project naming conventions, i.e. `Task`.

File: README.md (hunk 1 above)
Comment (tchaton) [reply README.md:122]: Done !

File: README.md (hunk 2 above)
Comment (tchaton) [reply README.md:125]: Changed for:

```
from flash.text import TranslationTask

model = TranslationTask.load_from_checkpoint("https://flash-weights.s3.amazonaws.com/translation_model_en_ro.pt")
model.serve()
```

File: flash/core/data/batch.py (hunk 3 above)
Comment (tchaton) [reply flash/core/data/batch.py:320]: Done ! Added a todo.

**Pull request merged**